### PR TITLE
fix(tests) : Added links,label and datatable tests

### PR DESCRIPTION
**Pull request by Raunak1203**

### CODE REVIEW

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

File: runtime/src/tests/functionalTests/page_objects/planner/planner.page.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ export class PlannerPage extends AppPage {
   quickAdd =  new planner.WorkItemQuickAdd($('alm-work-item-quick-add'));
   sidePanel = new planner.SidePanel($('aside.f8-sidepanel'));
   quickPreview = new planner.WorkItemQuickPreview($('work-item-preview'));
-
+  header = new planner.ToolbarHeader($('pfng-toolbar'));
+  settings = new planner.Settings($('div.f8-wi-list__settings'));
   constructor(url: string){
     super();
     this.url = url;
```
Comment (jarifibrahim) [inline runtime/src/tests/functionalTests/page_objects/planner/planner.page.ts]: We should change the constructor. Can you please change 
```
 constructor(url: string){
     super();
     this.url = url;
```
to
```
 constructor(url: string){
     super(url);
```

Comment (Raunak1203) [reply runtime/src/tests/functionalTests/page_objects/planner/planner.page.ts]: done

Comment (Raunak1203) [reply runtime/src/tests/functionalTests/page_objects/planner/planner.page.ts]: protractor config updated please check

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-4 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
@Raunak1203 Could any of these tests qualify (stable) to be included into smoketests?

**Comment by Raunak1203** [comment]
@rgarg1 now we will be running all the test in each PR.. Yes i ran all the test several times...It should be stable

**Comment by rgarg1** [comment]
Thats great!

**Comment by rgarg1** [comment]
I'll try running the tests tomorrow before approving. I hope this is fine with you.

**Comment by Raunak1203** [comment]
ok

File: runtime/src/tests/functionalTests/protractorTS.config.ts [hunk 2]
```diff
@@ -21,10 +21,15 @@ let conf: Config = {
   // Ref: https://github.com/angular/protractor/tree/master/exampleTypescript/asyncAwait
   SELENIUM_PROMISE_MANAGER: false,
 
-  specs: ["specs/*.js", "specs/**/*.js"],
-
+  specs: [
+    "specs/smoke/smokeTest.spec.js",
```
Comment (rgarg1) [inline runtime/src/tests/functionalTests/protractorTS.config.ts]: Why do we have mention to .js files in TS config?

Comment (Raunak1203) [reply runtime/src/tests/functionalTests/protractorTS.config.ts]: @rgarg1 typescript compiler will generate .js file which you can run

Comment (rgarg1) [reply runtime/src/tests/functionalTests/protractorTS.config.ts]: oh..okay. Thanks

Comment (Raunak1203) [reply runtime/src/tests/functionalTests/protractorTS.config.ts]: @rgarg1 now smokeTest are moved to a new folder smoke...So please delete smokeTest.spec.js inside functionalTests/specs/ and then try running.
smokeTest are moved to new folder to run smokeTest first in each PR and then other tests

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
@Raunak1203 

Tests run successfully after deleting all .js files from the directory. Good job!

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-10 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-11 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by nimishamukherjee** [comment]
@Raunak1203 the tests fails locally

1) Planner Smoke Tests: create a work item and add/remove assignee
  - Failed: stale element reference: element is not attached to the page document
    (Session info: chrome=64.0.3282.140)
    (Driver info: chromedriver=2.35.528139 (47ead77cb35ad2a9a83248b292151462a66cd881),platform=Linux 4.14.16-300.fc27.x86_64 x86_64)

2) Planner Smoke Tests: Check WorkItem creator name and image is reflected
  - Failed: stale element reference: element is not attached to the page document
    (Session info: chrome=64.0.3282.140)
    (Driver info: chromedriver=2.35.528139 (47ead77cb35ad2a9a83248b292151462a66cd881),platform=Linux 4.14.16-300.fc27.x86_64 x86_64)

3) Planner Smoke Tests: Associate/Re-associate workitem with an Iteration
  - Failed: stale element reference: element is not attached to the page document
    (Session info: chrome=64.0.3282.140)
    (Driver info: chromedriver=2.35.528139 (47ead77cb35ad2a9a83248b292151462a66cd881),platform=Linux 4.14.16-300.fc27.x86_64 x86_64)

4) Work Item datatable list work item should show updated title when switching from flat to tree view
  - Failed: stale element reference: element is not attached to the page document
    (Session info: chrome=64.0.3282.140)
    (Driver info: chromedriver=2.35.528139 (47ead77cb35ad2a9a83248b292151462a66cd881),platform=Linux 4.14.16-300.fc27.x86_64 x86_64)

Executed 17 of 17 specs (4 FAILED) in 2 mins 36 secs.

**Comment by Raunak1203** [comment]
@nimishamukherjee UntilAbsent() was throwing staleElement Exception...Now it should work

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-14 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-16 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-18 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-19 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-20 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-21 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-22 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Comment by Raunak1203** [comment]
@nimishamukherjee i think now we can merge , we have more than 5 successful builds.

**Comment by fabric8cd** [comment]
@Raunak1203 fabric8/fabric8-ui:SNAPSHOT-PR-2466-23 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2466-fabric8-planner.badger.fabric8.io

**Pull request merged**